docs/_posts/2019-03-05-go.md

```diff
@@ -0,0 +1,50 @@
+
+It also became difficult to consistently format our code, especially as our codebase increased in size and complexity. To fix this common problem, we added the `golint` automated tool into our workflow. Code linters run source code against an applied set of stylistic and syntax rules, raising flags when this structure is broken.  Linting has been an indispensable productivity tool for writing code and has even sped up our code review process.  
+
+Documentation is essential to growing and sustaining the community around our project. We add to and tweak our README and contributor guides to encourage end-user self-efficiency and success. This has proven to be a time-consuming undertaking because our codebase is constantly evolving and being refactored. To our benefit, Go offers the `godoc` documentation tool to dynamically create documentation. This tool parses the code, relevant comments, and structure to generate documentation that evolves as fast as our code does. 
+
+## … and beyond the toolchain 
+
+With Go, we can produce statically linked binaries that fully contain our applications and run them on different platforms. This capability supported our decision to move to Docker-based development for our building and testing.  
+
+Secretless is consumed as a Linux binary that is packaged inside a Docker image. That way, customers can easily get up and running once they grab our image. We also use Docker for our CI pipeline to build and publish our site, check styling and links, and run tests. We host these processes in Docker to ensure they receive the proper Go versioning for testing and to isolate their environments to ensure they always run in the same (and expected) way. 
+
+When it comes to testing, I have mixed feelings about Docker. On one hand, without it, we cannot properly test our code and on the other, Docker has added an extra layer of complexity that makes our project difficult to understand and unapproachable. We find Docker beneficial because the alternative is a large assortment of messy Bash scripts and boilerplate code. Furthermore, contributors no longer need to fuss with tooling in order to configure their environment correctly - all they need is Docker and they are ready to get rolling. But with this assumption, we have added an extra pre-requisite that ultimately limits those who can contribute. In terms of testing norms, Docker-based development has forced us to break the standard Go testing conventions. Our tests have become custom to our project, wrapping the known Go tests with Docker-specific code. Consequently, it may become initially unclear to contributors how our tests work.  
```

Comment:
it's worth digging into
> without it, we cannot properly test our code

people might rightfully wonder why? but we want to be able to run integration tests with Secretless running in its Docker container communicating with a database service, for example - and Docker is the easiest way to do this

Reply:
Can we swap the order of the positive and negative comments in this paragraph to end on a positive note? To me it reads better as

> When it comes to testing, I have mixed feelings about Docker. ~On one hand, without it, we cannot properly test our code and on the other,~ Docker has added an extra layer of complexity that [may make] our project ~difficult to understand and unapproachable~ less approachable. [By requiring Docker] we have added an extra pre-requisite that ~ultimately limits~ may limit those who can contribute. In terms of testing norms, Docker-based development has forced us to break the standard Go testing conventions. Our tests have become custom to our project, wrapping the known Go tests with Docker-specific code. Consequently, it may become initially unclear to contributors how our tests work.
> However, in using Docker we are able to run integration tests that mimic a more realistic Secretless deployment, with Secretless running in a container and communicating with a database service, for example. We find Docker beneficial because the alternative is a large assortment of messy Bash scripts and boilerplate code. Furthermore, contributors no longer need to fuss with tooling in order to configure their environment correctly - all they need is Docker and they are ready to get rolling.

Reply:
I also felt that ending on a negative note felt odd and happy you picked up on this as well.
@typaulhus please wait for @izgeri approval on this one before merging into PR

Reply:
see below @izgeri

Reply:
see below changes